Samples/BlazorExample/BlazorExample/BlazorExample.Client/Components/Pages/Counter.razor

```diff
@@ -1,35 +1,55 @@
 ﻿@page "/counter"
-@rendermode @(new InteractiveAutoRenderMode(prerender: false))
-@implements IDisposable /* CSLA wasm pages must be IDisposable */
+@* @rendermode @(new InteractiveAutoRenderMode(prerender: false)) *@
+@rendermode InteractiveAuto
 @inject Csla.Blazor.State.StateManager StateManager
+@inject ApplicationContext applicationContext
+@inject RenderModeProvider renderModeProvider
 
 <PageTitle>Counter</PageTitle>
 
 <h1>Counter</h1>
 
 <p role="status">Current count: @currentCount</p>
 
-<button class="btn btn-primary" @onclick="IncrementCount">Click me</button>
+<p class="text-info">@renderModeText</p>
+
+@if (IsInteractive)
+{
+    <button class="btn btn-primary" @onclick="IncrementCount">Click me</button>
+}
 
 @code {
     [CascadingParameter]
     private Task<AuthenticationState>? authenticationState { get; set; }
     private int currentCount = 0;
+    private bool IsInteractive = false;
+    private string? renderModeText;
 
     protected override async Task OnInitializedAsync()
     {
         // Every page _must_ initialize the state manager
         await StateManager.InitializeAsync();
+        var renderMode = renderModeProvider.GetRenderMode(this);
+        renderModeText = renderMode.ToString();
+        IsInteractive = renderMode == RenderModes.WasmInteractive || renderMode == RenderModes.ServerInteractive;
+
+        if (IsInteractive)
+        {
+            if (!applicationContext.LocalContext.Contains("counter"))
+            {
+                applicationContext.LocalContext["counter"] = 0;
+                StateManager.SaveState();
+            }
+            var tmp = applicationContext.LocalContext["counter"];
+            currentCount = tmp != null ? (int)tmp : 0;
```

Comment:
Do you expect concurrency issues here? Otherwise `LocalContext["counter"]` should never return `null`. You are making sure it's at least initialized with `0` 3 lines before.